include/sortcheck.h

```diff
@@ -199,6 +199,58 @@ inline bool binary_search_checked_full(_ForwardIterator __first,
   return binary_search_checked_full(__first, __last, __val, compare, file, line);
 }
 
+template<typename _ForwardIterator, typename _Tp, typename _Compare>
+inline _ForwardIterator lower_bound_checked(_ForwardIterator __first,
+                                  _ForwardIterator __last,
+                                  const _Tp &__val, _Compare __comp,
+                                  const char *file, int line) {
+  // Ordered
+  if (__first != __last) {
+    bool is_prev_less = true;
+    unsigned pos = 0;
+    for (_ForwardIterator it = __first; it != __last; ++it, ++pos) {
+      bool is_less = *it < __val;
+      if (is_less && !is_prev_less) {
+        std::ostringstream os;
+        os << file << ':' << line << ": unsorted range "
+           << "in position " << pos;
+        report_error(os.str());
+      }
+      is_prev_less = is_less;
+    }
+  }
```

Comment:
We should move this code to `check_sorted` to avoid code dup with `binary_search`.

Reply:
I say we should deal with it in a separate PR

Reply:
Do you have plans to implement this refactor?

Reply:
I've factored out the `check_ordered` code and fixed some bugs in it. Could you use it here instead of code dup?

Reply:
Done!